ansible_collections/arista/avd/roles/eos_designs/templates/network_services/logic.j2

```diff
@@ -1,32 +1,63 @@
 {% set tmp_tenants = [] %}
+{% set match_tags = switch.filter_tags %}
+{% if switch.group is arista.avd.defined %}
+{%     do match_tags.append(switch.group) %}
+{% endif %}
+{% set whitelist = namespace(vlans=switch.vlans | arista.avd.default([]) | arista.avd.range_expand | map('int')) %}
+{% if switch.uplink_type is arista.avd.defined('port-channel') %}
+{%     for uplink_interface in switch.uplink_interfaces | arista.avd.default([]) %}
+{%         set uplink_switch = switch.uplink_switches[loop.index0] | arista.avd.default %}
+{%         if uplink_switch is arista.avd.defined and uplink_switch in groups[fabric_name] %}
+{%             set uplink_switch_vlans = avd_switch_facts[uplink_switch].switch.vlans | arista.avd.default([]) | arista.avd.range_expand | map('int') %}
+{%             set whitelist.vlans = whitelist.vlans | select('in', uplink_switch_vlans) %}
+{%         endif %}
+{%     endfor %}
+{% endif %}
 {% set tmp_hostvars = hostvars[inventory_hostname] %}
 {% for network_services_key in network_services_keys | arista.avd.natural_sort('name') %}
 {%     if network_services_key.name is arista.avd.defined and tmp_hostvars[network_services_key.name] is arista.avd.defined %}
 {# tenants #}
 {%         for tenant in tmp_hostvars[network_services_key.name] | arista.avd.convert_dicts | arista.avd.natural_sort('name') %}
-{%             if tenant.name in switch.tenants | arista.avd.default([]) %}
```

Comment:
I have removed the `switch.tenants` "tree` to avoid carrying a large variable tree around in facts. So this logic was previously in `switch.j2` (look at the bottom), where it implemented the filtering as part of facts.

Reply:
ok - I did not read through all of switch.j2 during my review. Resolving this